docs/caching_spec.md

```diff
@@ -0,0 +1,99 @@
+# Trestle caching architecture
+
+## Table of Contents
+
+- [Purpose](<#purpose>)
+- [Users](<#users>)
+- [Caching Structure](<>)
+- [Dist vs editable structures](<>)
+- [Supported URIs](<>)
+- [Git support](<>)
+- [Update behaviour](<>)
+
+## Purpose
+
+This document describes the caching strategy and behaviour that is supported by trestle for 3rd party artifacts that are
+external to the core repo. A principle that trestle is expected to uphold is that documents, unless the user explicitly
+contradicts, are referred to by reference.
+
+For example the user should *not* be expected to manually copy NIST 800-53 into their working directory (although they may). They waould refer to the source document such that integrity is maintained. Given this a caching strategy is required to manage these documents - especially when operating in a CICD world.
+
+## Users
+
+The scope of this document is for developers. Once implemented additional documentation should be created for end users.
+
+## Scope
+
+This document is only concerned with links which refer to other OSCAL schema objects. External references (say to an xccdf document) are not considered.
+
+## Types of external references.
+
+NIST defines many types of referential links within the OSCAL schema. Here trestle is interested in supporting a subset of these links. In particular these links are characterized specifically by the fact that they refer to 'upstream' OSCAL
+artifacts. The table below identifies the references covered by this document.
+
+| Schema               | Json path (\[*\] represents array or named keys)                      | Refers to            |
+| \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-| \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-- | \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\--- |
+| profile              | profile.imports\[*\].href                                             | catalog or profile   |
```

Comment:
Should we include `profile.merge` and `profile.modify`?

Reply:
profile.merge and profile.modify do not have external hrefs. Controls must first be imported.